In [29]:
! pip install yfinance 

# The Spot Rate Cross Exchange Rates and Percentage Changes

In [66]:
import yfinance as yf
import pandas as pd 
import numpy as np
from datetime import datetime, timedelta

In [67]:
Ticker = 'USDEUR=X'
start_date = '2023-01-01'
end_date= '2023-12-31'

In [68]:
data = yf.download(Ticker, start=start_date, end=end_date)

[*********************100%%**********************]  1 of 1 completed


In [69]:
data

Open     High       Low    Close  Adj Close  Volume
Date                                                              
2023-01-02  0.93373  0.93868  0.933500  0.93373    0.93373       0
2023-01-03  0.93653  0.95043  0.936100  0.93653    0.93653       0
2023-01-04  0.94815  0.94823  0.940600  0.94815    0.94815       0
2023-01-05  0.94283  0.95097  0.940500  0.94283    0.94283       0
2023-01-06  0.95037  0.95372  0.941420  0.95037    0.95037       0
...             ...      ...       ...      ...        ...     ...
2023-10-25  0.94381  0.94636  0.942800  0.94381    0.94381       0
2023-10-26  0.94632  0.95000  0.946100  0.94632    0.94632       0
2023-10-27  0.94657  0.94910  0.943600  0.94657    0.94657       0
2023-10-30  0.94675  0.94801  0.941088  0.94675    0.94675       0
2023-10-31  0.94180  0.94400  0.937100  0.93770    0.93770       0

[217 rows x 6 columns]

In [65]:
end_date_0 = datetime.now()
start_date_0 = end_date_0 - timedelta(days=365)

In [35]:
exchange_rate_date_0 = yf.download(Ticker, start=start_date_0, end=end_date_0)

[*********************100%%**********************]  1 of 1 completed


In [36]:
spot_exchange_rate_0 = exchange_rate_date_0['Adj Close'].iloc[-1]

In [37]:
spot_exchange_rate_0

0.9404000043869019

In [38]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_0 = 0.035 # Example: 1% nominal interest rate in EUR
nominal_interest_rate_foreign_0 = 0.052 # Example: 2% nominal interest rate in USD


In [39]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate = exchange_rate_date_0['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate = current_exchange_rate * (1 + nominal_interest_rate_foreign_0) / (1 + nominal_interest_rate_home_0)


In [40]:
print(f"Expected future USD/EUR exchange rate: {expected_exchange_rate:.4f}")

Expected future USD/EUR exchange rate: 1.0210


In [41]:
Ticker_1 = 'CHFUSD=X'
start_date_1 = '2023-01-01'
end_date_1 ='2023-12-31'

In [42]:
data_1 = yf.download(Ticker_1, start=start_date_1, end=end_date_1)

[*********************100%%**********************]  1 of 1 completed


In [43]:
data_1.tail()

Open      High       Low     Close  Adj Close  Volume
Date                                                                 
2023-10-25  1.119746  1.121202  1.115573  1.119746   1.119746       0
2023-10-26  1.115200  1.115573  1.111198  1.115200   1.115200       0
2023-10-27  1.112557  1.113586  1.106929  1.112557   1.112557       0
2023-10-30  1.108156  1.110494  1.105278  1.108156   1.108156       0
2023-10-31  1.109225  1.109755  1.107211  1.109275   1.109275       0

In [44]:
end_date_1 = datetime.now()
start_date_1 = end_date_1 - timedelta(days=365)

In [45]:
exchange_rate_date_1 = yf.download(Ticker_1, start=start_date_1, end=end_date_1)

[*********************100%%**********************]  1 of 1 completed


In [46]:
spot_exchange_rate_1 = exchange_rate_date_1['Adj Close'].iloc[-1]

In [47]:
spot_exchange_rate_1

1.1092623472213745

In [48]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_1 = 0.017  # Example: 1% nominal interest rate in CHF
nominal_interest_rate_foreign_1 = 0.052  # Example: 2% nominal interest rate in USD

In [49]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate_1 = exchange_rate_date_1['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate_1 = current_exchange_rate * (1 + nominal_interest_rate_home_1) / (1 + nominal_interest_rate_foreign_1)

In [50]:
print(f"Expected future CHF/EUR exchange rate: {expected_exchange_rate_1:.4f}")

Expected future CHF/EUR exchange rate: 0.9711


In [51]:
Ticker_2 = 'USDGBP=X'
start_date_2 = '2023-01-01'
end_date_2 = '2023-12-31'

In [52]:
data_2 = yf.download(Ticker_2, start=start_date_2, end=end_date_2)

[*********************100%%**********************]  1 of 1 completed


In [53]:
data_2.tail()

Open     High      Low    Close  Adj Close  Volume
Date                                                             
2023-10-25  0.82215  0.82538  0.82120  0.82224    0.82224       0
2023-10-26  0.82630  0.82847  0.82427  0.82621    0.82621       0
2023-10-27  0.82399  0.82600  0.82225  0.82392    0.82392       0
2023-10-30  0.82555  0.82695  0.82197  0.82551    0.82551       0
2023-10-31  0.82186  0.82382  0.82171  0.82233    0.82233       0

In [54]:
end_date_2 = datetime.now()
start_date_2 = end_date_2 - timedelta(days=365)

In [55]:
exchange_rate_date_2 = yf.download(Ticker_2, start=start_date_2, end=end_date_2)

[*********************100%%**********************]  1 of 1 completed


In [56]:
spot_exchange_rate_2 = exchange_rate_date_2['Adj Close'].iloc[-1]

In [57]:
spot_exchange_rate_2

0.8222000002861023

In [58]:
# Define nominal interest rates for the home (EUR) and foreign (USD) countries
nominal_interest_rate_home_2 = 0.052  # Example: 1% nominal interest rate in GBP
nominal_interest_rate_foreign_2 = 0.052  # Example: 2% nominal interest rate in USD

In [59]:
# Calculate the expected future exchange rate (EUR/USD) using the IFE formula
current_exchange_rate_2 = exchange_rate_date_2['Adj Close'].iloc[0]  # Use the first day's rate
expected_exchange_rate_2 = current_exchange_rate * (1 + nominal_interest_rate_home_2) / (1 + nominal_interest_rate_foreign_2)

In [60]:
print(f"Expected future USD/GBP exchange rate: {expected_exchange_rate_2:.4f}")

Expected future USD/GBP exchange rate: 1.0045


The spot rate CHF/EUR cross-rate is closest to USD/EUR * CHF/USD = 0.9404 * 1.1092 = 1.0430

The spot rate GBP/EUR cross-rate is closest to USD/EUR * (USD/GBP)*-1 = USD/EUR / GBP/USD = 0.9404/0.8222=1.1437

 The euro is the base currency in the USD/EUR quote, and the 
expected decrease in the USD/EUR rate indicates that the EUR is depreciating 
(in one year it will cost less USD to buy one EUR). = 1.0210/0.9404 -1 = 0.0857

The GBP is the base currency in the USD/GBP quote, and the 
expected decrease in the USD/GBP rate means that the GBP is expected to 
depreciate against the USD. Or equivalently, the USD is expected to appreciate 
against the GBP. Mathematically = (1.0045/0.8218)*-1 = 0.8222/1.0045-1=-0.1814

. According to the table, USD/EUR is expected to decline from 0.9404
to 1.0210, while USD/GBP is expected to decline from 0.8222 to 1.0045. So, the 
USD is expected to be stronger than both the EUR and GBP. GBP/EUR is currently 1.1437 [= (0.8222)–1 × 0.9404] and is expected to be 1.0164 [= (1.0045)–1 × 1.0210 ], so the GBP is expected to be stronger than the EUR. 